modules/data_models.py

```diff
@@ -0,0 +1,363 @@
+        """
+        self.line = line
+        self.populate()
+
+    def populate(self):
+        """
+        Sets the values of highest_temperature, lowest_temperature, max_humidity,
+        mean_humidity and date by using the parsed_line
+        Returns:
+            None
+        """
+        self.highest_temperature = get_highest_temperature(self.line)
+        self.lowest_temperature = get_lowest_temperature(self.line)
+        self.max_humidity = get_max_humidity(self.line)
+        self.mean_humidity = get_mean_humidity(self.line)
+        self.date = get_date(self.line)
+
+    def get_max_temperature(self):
+        """
+        Returns the highest_temperature, can be None
+        Returns:
+            (int or None):  highest_temperature in degree celsius
+                            Or
+                            None
+        """
+
+        return self.highest_temperature
+
+    def get_min_temperature(self):
+        """
+        Returns the lowest_temperature, can be None
+        Returns:
+            (int or None):  lowest_temperature in degree celsius
+                            Or
+                            None
+        """
+
+        return self.lowest_temperature
+
+    def get_max_humidity(self):
+        """
+        Returns the max_humidity, can be None
+        Returns:
+            (int or None):  max_humidity in percentage
+                            Or
+                            None
+        """
+
+        return self.max_humidity
+
+    def get_mean_humidity(self):
+        """
+        Returns the mean_humidity, can be None
+        Returns:
+            (int or None):  mean_humidity in percentage
+                            Or
+                            None
+        """
+
+        return self.mean_humidity
+
+    def get_date(self):
+        """
+        Returns the date
+        Returns:
+            (str): Value containing date in yyyy-mm-dd format e.g '2005-6-2'
+        """
+
+        return self.date
+
+
+class MonthData:
+    """
+    This class holds the data of an entire month
+    """
+
+    def __init__(self, year, month, path=WEATHER_FILES_DIR):
+        """
+        Initializes the members name, year and path and
+        calls populate to fill days_data list
+        Args:
+            year(str or int):   Value containing 4 digit year e.g '2004'
+            month(str): Value containing 3 character month name e.g 'Feb', 'Mar'
+            path(str):  Value containing path to weather files e.g 'weatherfiles/'
+        """
+        self.days_data = []
+        self.name = month
+        self.year = year
+        self.path = path
+        self.populate()
+
+    def populate(self):
+        """
+        Initializes the days_data list with DayData objects for each day of the month
+        Returns:
+            None
+        """
+
+        month_data = next(read_data(f"{self.year}_{self.name[0:3]}", self.path))
+        for line in month_data:
+            day_data = DayData(parse_line(line))
```

Comment:
We should just pass the `line` here, the `parse_line(line)` should be called from within the DayData Class in the populate. MonthData doesn't need to parse it for the day, The `DayaData` can use the `utils` independently and parse the line itself.

Reply:
Done